icu4j/maven-build/maven-icu4j/pom.xml

```diff
@@ -0,0 +1,493 @@
+            </goals>
+            <configuration>
+              <resources>
+                <resource>
+                  <directory>../../main/tests/collate/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+                <resource>
+                  <directory>../../main/tests/core/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+                <resource>
+                  <directory>../../main/tests/framework/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+                <resource>
+                  <directory>../../main/tests/packaging/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+                <resource>
+                  <directory>../../main/tests/translit/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+                <resource>
+                  <directory>../../tools/misc/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+
+
+        </executions>
+      </plugin>
+
+      <!--
+        Extract and copy the *.res files from the *.jar files in the upstream depenendency `com.ibm.icu:icu4j-datafiles`.
+        Enables the creation of *.lst files in the next phase using the helper script.
+        Use the ./target/classes directory as the output directory so that the files (included generated *.lst files)
+        get included in the shaded .jar file.
+      -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>3.3.0</version>
+        <executions>
+          <execution>
+            <id>unpack-dependencies</id>
+            <phase>generate-resources</phase>
+            <goals>
+              <goal>unpack-dependencies</goal>
+            </goals>
+            <configuration>
+              <includeGroupIds>com.ibm.icu</includeGroupIds>
+              <!-- exclude test resource files which come through dependencies as `provided` scope -->
+              <excludeScope>provided</excludeScope>
+              <!--
+                Don't know why the Ant build.xml is configured to exclude `**/res_index.res` from the set of
+                included files since those files still appear in the packaged .jar file.
+              -->
+              <excludes>**/*.cnv,**/cnvalias.icu</excludes>
```

Comment:
maybe. IIRC icu4j uses its own separate index built a different way.

Reply:
I've been comparing the .jar file contents between the Ant build and the Maven build for the 3 artifacts: icu4j, icu4j-charset, and icu4j-localespi.  Except for minor discrepancies like Ant generating a `package.html` per directory of class files, they should align.

We need more scrutiny on what is in the .jar files and deciding what we actually need.